README.md

```diff
@@ -4,6 +4,41 @@ This allows us to perform the actual codesigning steps necessary to release MacO
 
 See [here](https://brevi.link/design-code-signing) for the general design that this fits into.
 
-## Example use
+## Overview
 
-TODO
+This plugin relies upon the build agent already having the necessary keychains created on the machine.
+
+Unfortunately, this is a necessity due to macos' insistence upon a one-time-per-key/keychain manual
+intervention to approve access to the signing key before `codesign` can use it (even if the cert/key are imported with `codesign` pre-granted access to use it).
+
+## Example use TODO(seanrobertson)
+
+```yaml
+- label: "sign-macos-binary"
+  command: "" # No command needed here, since the `command` hook does the work.
+  agents:
+    - "queue=macos-codesigner"
+  artifact_paths: "signed-thing.bin"
+  plugins:
+    - mac-codesign#v1.0.0:
+        input_artifact: "unsigned-thing.bin"
+        keychain: "production-certs.keychain"
+        keychain_secret: "secret/sync.v1/dev-workflow/production-buildkite/buildkite-agents/cert-decryption-password/ci/improbable/production-codesigning"
```

Comment:
Maybe `keychain_unlocker_vault_path`

Reply:
I went with keychain_unlock_secret_name, since I also turned this into just the _name_ of the secret instead of the whole path.